processing/src/main/java/io/druid/query/timeboundary/TimeBoundaryQuery.java

```diff
@@ -120,16 +129,19 @@ public TimeBoundaryQuery withQuerySegmentSpec(QuerySegmentSpec spec)
         dataSource,
         getQuerySegmentSpec(),
         bound,
+        dimFilter,
         getContext()
     );
   }
 
   public byte[] getCacheKey()
   {
+    final byte[] filterBytes = dimFilter == null ? new byte[]{} : dimFilter.getCacheKey();
     final byte[] boundBytes = StringUtils.toUtf8(bound);
     return ByteBuffer.allocate(1 + boundBytes.length)
```

Comment:
Couple points here:

1. This ByteBuffer needs to be larger to accommodate `filterBytes`.
2. Typically we put `0xff` after a string like `boundBytes` in the cache key to delimit it from other parts of the key